### PR TITLE
add tristate verification to twitch

### PR DESCRIPTION
**Pull request by 0x1**
### Description:
add tristate verification to twitch

### Checklist:
* [ ] Tests passing (`make test-community`)?
* [ ] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/usage/install/#local-installation))?

### CODE REVIEW

File: pkg/detectors/twitch/twitch_test.go [hunk 1]
```diff
@@ -47,14 +49,87 @@ func TestTwitch_FromChunk(t *testing.T) {
...
 					Verified:     true,
 				},
+				{
+					DetectorType: detectorspb.DetectorType_Twitch,
+					Verified:     false,
+				},
+				{
+					DetectorType: detectorspb.DetectorType_Twitch,
+					Verified:     false,
+				},
```
Comment (rosecodym) [inline pkg/detectors/twitch/twitch_test.go:71]: is the order of these results consistent? i think that the test will flake if not.

Comment (0x1) [reply pkg/detectors/twitch/twitch_test.go:71]: yes it should always consistent bc the data string doesn't change.

File: pkg/detectors/twitch/twitch.go [hunk 2]
```diff
@@ -58,37 +62,59 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
...
+
+	req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+	req.Header.Add("Content-Length", strconv.Itoa(len(data.Encode())))
+	res, err := client.Do(req)
+	if err != nil {
+		return false, err
+	}
+
+	defer res.Body.Close()
+	if res.StatusCode == http.StatusOK {
```
Comment (ahrav) [inline pkg/detectors/twitch/twitch.go:109]: nit (optional): You could switchify this and the default case handles the final return.
```
switch res.StatusCode {
case http.StatusOK:
	return true, nil
case http.StatusBadRequest, http.StatusForbidden:
	return false, nil
default:
	return false, fmt.Errorf("unexpected http response status %d", res.StatusCode)
```

**Review by ahrav — APPROVED** [review]

File: pkg/detectors/twitch/twitch.go [hunk 3]
```diff
@@ -13,13 +14,16 @@ import (
...
+	client *http.Client
+}
 
 // Ensure the Scanner satisfies the interface at compile time
 var _ detectors.Detector = (*Scanner)(nil)
 
 var (
-	client = common.SaneHttpClient()
+	defaultClient = common.SaneHttpClient()
+	verifyURL     = "https://id.twitch.tv/oauth2/token"
```
[the remark this answers is not in the document]
Comment (0x1) [reply pkg/detectors/twitch/twitch.go:26]: u right, prob should be const

**Pull request merged**